allennlp/experiments/drivers/train_driver.py

```diff
@@ -0,0 +1,182 @@
+from typing import Optional
+import logging
+import os
+import shutil
+
+import torch
+import tqdm
+from overrides import overrides
+
+from allennlp.common import Params
+from allennlp.common.checks import ConfigurationError
+from allennlp.experiments.driver import Driver
+from allennlp.data.data_iterator import DataIterator
+from allennlp.training import Model
+from allennlp.training.optimizers import get_optimizer_from_params
+from allennlp.data import Dataset, Vocabulary
+from allennlp.data.dataset_reader import DatasetReader
+from allennlp.experiments.registry import Registry
+from allennlp.common.tensor import arrays_to_variables
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@Registry.register_driver("train")
+class TrainDriver(Driver):
+    def __init__(self,
+                 model: Model,
+                 optimizer: torch.optim.Optimizer,
+                 iterator: DataIterator,
+                 train_dataset: Dataset,
+                 validation_dataset: Optional[Dataset] = None,
+                 patience: int = 2,
+                 batch_size: int = 32,
```

Comment:
This isn't a trainer parameter, it's a data iterator parameter.  If you think it makes more sense as a trainer parameter, we should think about changing the iterator API.